corehq/util/couch_doc_processor.py

```diff
@@ -1,16 +1,39 @@
-import hashlib
 import weakref
 from abc import ABCMeta, abstractproperty
 from datetime import datetime
 
 import six
 from couchdbkit import ResourceNotFound
 
-from corehq.util.couch_helpers import paginate_view
-from corehq.util.pagination import PaginationEventHandler, DelegatingPaginationEventHandler
+from corehq.util.couch_helpers import PaginatedViewArgsProvider
+from corehq.util.pagination import PaginationEventHandler, ResumableFunctionIterator, TooManyRetries
 
 
-class ResumableDocsByTypeIterator(object):
+class ResumableDocsByTypeArgsProvider(PaginatedViewArgsProvider):
+    def __init__(self, initial_view_kwargs, doc_types):
+        super(ResumableDocsByTypeArgsProvider, self).__init__(initial_view_kwargs)
+        self.doc_types = doc_types
```

Comment:
Should probably be `self.doc_types = list(doc_types)` (make a copy) since it gets mutated.